src/uct/base/uct_mem.c

```diff
@@ -114,40 +103,65 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
 
                 /* Check if MD supports allocation with fixed address
                  * if it's requested */
-                if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
+                if ((param->flags & UCT_MD_MEM_FLAG_FIXED) &&
                     !(md_attr.cap.flags & UCT_MD_FLAG_FIXED)) {
                     continue;
                 }
 
+                /* Check if MD supports allocation on requested mem_type */
+                if (!(md_attr.cap.alloc_mem_types & param->mem_type)) {
```

Comment:
wrong & check

Reply:
Thanks for pointing this. Fixed some of the errors.